src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -25,252 +26,1010 @@
 #include "precompiled.hpp"
 #include "gc/shared/tlab_globals.hpp"
 #include "gc/shenandoah/shenandoahFreeSet.hpp"
+#include "gc/shenandoah/shenandoahFreeSet.inline.hpp"
 #include "gc/shenandoah/shenandoahHeap.inline.hpp"
 #include "gc/shenandoah/shenandoahHeapRegionSet.hpp"
 #include "gc/shenandoah/shenandoahMarkingContext.inline.hpp"
 #include "logging/logStream.hpp"
 #include "memory/resourceArea.hpp"
 #include "runtime/orderAccess.hpp"
 
-ShenandoahFreeSet::ShenandoahFreeSet(ShenandoahHeap* heap, size_t max_regions) :
-  _heap(heap),
-  _mutator_free_bitmap(max_regions, mtGC),
-  _collector_free_bitmap(max_regions, mtGC),
-  _max(max_regions)
+static const char* partition_name(ShenandoahFreeSetPartitionId t) {
+  switch (t) {
+    case NotFree: return "NotFree";
+    case Mutator: return "Mutator";
+    case Collector: return "Collector";
+    default: return "Unrecognized";
```

Comment:
I believe using an 'enum class' for ShenandoahFreeSetPartitionId means you don't need to have a default.

Reply:
Compiler still wants the default clause.  Otherwise, "control reaches end of non-void function".  You'd think the compiler could be less strict, but will leave as is, unless we can figure out a way to get around this.